pkg/cluster/serviceprincipal.go

```diff
@@ -54,7 +54,7 @@ func (m *manager) clusterSPObjectID(ctx context.Context) error {
 }
 
 func (m *manager) fixupClusterSPObjectID(ctx context.Context) error {
-	if m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.SPObjectID != "" {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
```

Comment:
This shouldn't be changed, this field is not related to MIWI cluster installation, I believe this function call will be handled for the MIWI flow as part of Update story.